turbine/src/cluster_nodes.rs

```diff
@@ -363,6 +377,46 @@ fn cmp_nodes_stake(a: &Node, b: &Node) -> Ordering {
         })
 }
 
+// Dedups socket addresses so that if there are 2 nodes in the cluster with the
+// same TVU socket-addr, we only send shreds to one of them.
+// Also limits number of nodes at the same IP address.
+fn dedup_tvu_addrs(nodes: &mut Vec<Node>) {
+    const TVU_PROTOCOLS: [Protocol; 2] = [Protocol::QUIC, Protocol::UDP];
+    let capacity = nodes.len().saturating_mul(2);
+    // Tracks (Protocol, SocketAddr) tuples already observed.
+    let mut addrs = HashSet::with_capacity(capacity);
+    // Maps IP addresses to number of nodes at that IP address.
+    let mut counts = HashMap::with_capacity(capacity);
```

Comment:
Is it wise to reallocate 2 HashMaps for every shred here? With default fanout of 200, these will get 400 elements each, that is 800 elements total. With larger fanouts this will get quite big. Is it wise to have this happen on a per shred basis or should this be optimized eventually?

Reply:
The output here ultimately goes to `ClusterNodesCache` which cache cluster-nodes with 5 seconds TTL.
So this is not per shred. It is once every 5 seconds.